libraries/botbuilder-core/botbuilder/core/bot_adapter.py

```diff
@@ -77,6 +78,19 @@ def use(self, middleware):
         BotAssert.context_not_none(context)
 
         if context.activity is not None:
+
+            # Middleware to assign tenant_id from channelData to Conversation.tenant_id.
+            # MS Teams currently sends the tenant ID in channelData and the correct behavior is to expose 
+            # this value in Activity.Conversation.tenant_id.
+            # This code copies the tenant ID from channelData to Activity.Conversation.tenant_id.
+            # Once MS Teams sends the tenant_id in the Conversation property, this middleware can be removed.
+            if (Channels.ms_teams == context.activity.channel_id and context.activity.conversation is not None
```

Comment:
this should really only be in the path in BotFrameworkAdapter... in p[rocess_activity in the derived class